src/mo_frontend/typing.ml

```diff
@@ -2048,10 +2049,24 @@ and detect_lost_fields env t = function
          | Some _ -> ()
```

Comment:
Don't use List.find_opt on previous line (see below)

Reply:
Good point, thanks!

Reply:
Well thinking about this if the type has type fields, the with record (extension) it will always cause a type error, consider:
``` Motoko
let foo_type_warn : { type Quux = Text; foo : Text} = { foo = "FOO"; Quux = "QUUX" };
```
since we can't define type fields with records...

Reply:
Fixed the code anyway!